### PR TITLE
Add notification window when pressed on title bar and info-box, Add new TFT70 Status Screen, Fix FAN_CTRL_COUNT error during config upload & clean up

### LINKED ISSUE #1064: error in config.ini
### Description

updating config.ini gives an error on the LCD saying: "invalid Values.  fan_ctrl_count: 0"
changed value to 1 but error remains

### Steps to reproduce

update to latest firmware, self compiled

**Expected behavior**
no errors
**Actual behavior**
all works after update but not sure if I will encounter problems in the near future.
Did not do a test print yet

### Hardware Variant
tft 3,5 v3
### TFT Firmware Version & Main Board Firmware details
Latest marlin

### PR DESCRIPTION
**Pull request by guruathwal**
- Add notification window to show last 3 messages received from the printer/machine when pressed on Title Bar on any screen and the Info-box on the Status screen.
- Show notification dot in the top left corner when notifications are received.
- Add truncate option in `GUI_DispLenString` to truncate long text and avoid text clipping.
- Fix `fan_ctrl_count` error during config upload (#1064)
- Add a new status screen for TFT70
- BMP images on SPIflash are now stored with dimensions.
- Clean-up and remove unnecessary code and remove some global variables to reduce RAM usage.
- Add auto-detect for chamber heater and machine size (needs enabling M115_GEOMETRY_REPORT in marlin firmware).
- Add flash data/config check to notify if updating font, icons or config is required after a firmware update.

Fixes #1064

### CODE REVIEW

**Comment by traffic-light** [comment]
Nice. Was planning to do this by myself 😊

**Comment by oldman4U** [comment]
Gurmeet is my hero;-)

**Comment by blueeagle69** [comment]
Come on BTT merge this to the Master branch.

**Comment by traffic-light** [comment]
> Come on BTT merge this to the Master branch.

Especially because it's fixing a bug what's pretty annoying.

File: TFT/src/User/API/Language/language_en.h [hunk 1]
```diff
@@ -57,6 +57,8 @@
     #define EN_RESTORE                "Restore"
     #define EN_RESET                  "Reset"
     #define EN_DEFAULT                "Default"
+    #define EN_CLEAR                  "Clear"
+    #define EN_CLEAR                  "Clear"
```
Comment (radek8) [inline TFT/src/User/API/Language/language_en.h:61]: this is duplication

**Comment by guruathwal** [comment]
The fonts on TFT28 & TFT35 status screen were not meant to be larger.

**Comment by guruathwal** [comment]
if the preheat menu is freezing you may reflash the config file

**Comment by oldman4U** [comment]
Wouldn‘t it be good to have a kind of firmware vs config.ini check with a mismatch warning!?

**Comment by guruathwal** [comment]
@oldman4U  
almost done...
image

**Comment by oldman4U** [comment]
👍🏻

**Comment by guruathwal** [comment]
@discip you config file is not being read by the TFT. Try with the latest commit.

**Comment by guruathwal** [comment]
@GregSKR are you using the precompiled binaries or compiling yourself?

**Comment by GregSKR** [comment]
@guruathwal this is what I **always** do:
- download lastest version on [your fork](https://github.com/guruathwal/BIGTREETECH-TouchScreenFirmware/tree/master)
- compile it with Visual Studio Code
- change platformio.ini for TFT35V2.0 and config.ini for my personal parameters
- do a full flash (firmware+fonts+bmp+config.ini)

This is what I did a few hours ago, config.ini looks like crashing at loading...
Just did it again now, with your lastest commit and
- preheat menu is now ok, all labels displayed, no freezing 👍 
- the fonts on status screen are back to normal 👍 
- (and custom menus are ok, too)
THANKS FOR THIS

But still no notification history...
When power my Ender-3, I got 2 notifications on my TFT (**echo: fade height 10.0** and **echo: bed levelling ON**).
Go to Terminal and run **M117 TEST**
Back to status screen **echo: TEST** is showing
But history is still empty... 😢

**Comment by discip** [comment]
@guruathwal 
# THANK YOU! 👍
I second what @GregSKR stated above.
The good and the none working as well.

**Comment by oldman4U** [comment]
I believe i could find some time tomorrow and do some tests.

Got a brand new SKR v1.4 and have TFT 35 E3 and TFT 28 to test, as well as SKR E3 DIP and SKR Pro v1.1.

Any particular combination i should test?

Let me know

**Comment by guruathwal** [comment]
@GregSKR those messages are ignored by the firmware and only messages received from the printer are stored in the notification menu.
try sending messages with M118

**Comment by GregSKR** [comment]
@guruathwal Oh, my bad, didn't notice there is some filtering! I just tested with M118 and... got it!!!
Thanks again, looks like this PR is ready for merging ❤️

**Comment by digant73** [comment]
@guruathwal 
hi, in the "hostActionCommands" function on parseAck.c I will also change the dialog type to DIALOG_TYPE_ALERT for the three dialog boxes below (starting at line 236):

        BUZZER_PLAY(sound_notify);
        popupReminder(**DIALOG_TYPE_INFO**,(u8 *)"Message", (u8 *)hostAction.prompt_begin);
        break;
      case 1:
        BUZZER_PLAY(sound_notify);
        showDialog(**DIALOG_TYPE_ERROR**, (u8*)"Action command", (u8 *)hostAction.prompt_begin, (u8 *)hostAction.prompt_button1, NULL, breakAndContinue, NULL, NULL);
        break;
      case 2:
        BUZZER_PLAY(sound_notify);
        showDialog(**DIALOG_TYPE_ERROR**, (u8*)"Action command", (u8 *)hostAction.prompt_begin, (u8 *)hostAction.prompt_button1, (u8 *)hostAction.prompt_button2, resumeAndPurge, resumeAndContinue, NULL);

**Comment by traffic-light** [comment]
This is a promising update. Thank you for your time @guruathwal :-)

(the font/ icon/ ini check is really a great thing :-) )

**Comment by GregSKR** [comment]
Just saw then screen capture here: https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/1119#issuecomment-696946291

Looks like the 'Settings.c' code needs a few adjustments... 😊 
- missing a **ypos** increment (**ypos += BYTE_HEIGHT;**) after the icons check (so after line 266)
- shouldn't this increment be AFTER the curly braces (lines **237**, **246** and **256**) and not in the 'else' condition?

**Comment by charakaSamawry** [comment]
line 21 "REPORT_FAN_CHANGE (in Configuration_adv.h)" referring to line 3114 of config_adv (version 020006) seems to be irrelevant now..? (version 020008) guessing it doesnt matter but confusing and should be removed from top of config.ini, where it says you must enable it.

**Comment by guruathwal** [comment]
@charakaSamawry  it is still relevant. why is it confusing to enable a line in marlin firmware?

**Comment by charakaSamawry** [comment]
In version 20008 that line does not exist. 

> @charakaSamawry it is still relevant. why is it confusing to enable a line in marlin firmware?

**Comment by guruathwal** [comment]
It does
https://github.com/MarlinFirmware/Marlin/blob/50ba20fe1f27cdcb0f0f98f11cb42248dbc186f7/Marlin/Configuration_adv.h#L3205-L3207

**Comment by charakaSamawry** [comment]
> It does
> https://github.com/MarlinFirmware/Marlin/blob/50ba20fe1f27cdcb0f0f98f11cb42248dbc186f7/Marlin/Configuration_adv.h#L3205-L3207

So it does thank you. Weird I did search it in vscode .......no caps maybe 🤷‍♂️didn’t think that mattered though

**Comment by charakaSamawry** [comment]
im still confused lol! i dont have the line added on july 15th. but the change shows in my fork on july 15th. My fork also shows as  uptodate with bugfix .. i mustve buggered up somewhere i guess

**Comment by charakaSamawry** [comment]
https://github.com/charakaSamawry/Marlin/blob/4f50b68363c5779e547d323056d89713cf8b3a0b/Marlin/Configuration_adv.h

🤷‍♂️ lost some 'S's in http:// links when i changed some config stuff as well... sorry im a wolly

**Pull request merged**